File: src/main/java/me/brendn/oak/api/common/world/World.java
package me.brendn.oak.api.common.world;

import me.brendn.oak.api.common.entity.Entity;

import java.util.List;

/**
 * Represents a World.
 */
public interface World extends BlockAccess {

	/**
	 * @return The World's <a href="http://minecraft.gamepedia.com/Seed_(level_generation)">seed</a>.
	 */
	long getSeed();

	/**
	 * @return Whether or not it is currently raining.
	 */
	boolean isRaining();

	/**
	 * Returns all of the loaded entities.
	 *
	 * @return Loaded entities in the world
	 */
	List<Entity> getLoadedEntities();
}

File: src/main/kotlin/me/brendn/oak/api/common/world/World.kt
package me.brendn.oak.api.common.world

import me.brendn.oak.api.common.entity.Entity
import me.brendn.oak.api.common.world.biome.Biome

interface World : BlockAccess {

	/**
	 * @return The World's [seed](http://minecraft.gamepedia.com/Seed_(level_generation)).
	 */
	fun getSeed(): Long

	/**
	 * @return Whether or not it is currently raining.
	 */
	fun isRaining(): Boolean

	/**
	 * @return Loaded entities in the world
	 */
	fun getLoadedEntities(): Collection<Entity>

	fun getBiome(chunkX: Int, chunkZ: Int): Biome
}